src/tools/wasm-fuzz-lattices.cpp

```diff
@@ -0,0 +1,389 @@
+
+#include <optional>
+#include <random>
+#include <string>
+
+#include "analysis/lattice.h"
+#include "analysis/liveness-transfer-function.h"
+#include "analysis/reaching-definitions-transfer-function.h"
+
+#include "support/command-line.h"
+#include "tools/fuzzing.h"
+#include "tools/fuzzing/random.h"
+
+namespace wasm {
+using RandEngine = std::mt19937_64;
+
+uint64_t getSeed() {
+  // Return a (truly) random 64-bit value.
+  std::random_device rand;
+  return std::uniform_int_distribution<uint64_t>{}(rand);
+}
+
+struct Fuzzer {
+  bool verbose;
+
+  Random rand;
+
+  Fuzzer(bool verbose) : verbose(verbose), rand({}) {}
+
+  // Checks reflexivity of a lattice element, i.e. x = x.
+  template<typename Lattice>
+  void checkReflexivity(Lattice& lattice, typename Lattice::Element& element) {
+    analysis::LatticeComparison result = lattice.compare(element, element);
```

Comment:
It would be fine to stick a `using analysis;` at the beginning of this file so you don't need to rewrite the namespace everywhere.